src/OpenTelemetry.Contrib.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -0,0 +1,104 @@
+﻿// <copyright file="RuntimeMetrics.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics.Metrics;
+using System.Reflection;
+using System.Threading;
+
+namespace OpenTelemetry.Contrib.Instrumentation.Runtime
+{
+    /// <summary>
+    /// .NET runtime instrumentation.
+    /// </summary>
+    internal class RuntimeMetrics : IDisposable
+    {
+        internal static readonly AssemblyName AssemblyName = typeof(RuntimeMetrics).Assembly.GetName();
+        internal static readonly string InstrumentationName = AssemblyName.Name;
+        internal static readonly string InstrumentationVersion = AssemblyName.Version.ToString();
+
+        private readonly Meter meter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RuntimeMetrics"/> class.
+        /// </summary>
+        /// <param name="options">The options to define the metrics.</param>
+        public RuntimeMetrics(RuntimeMetricsOptions options)
+        {
+            this.meter = new Meter(InstrumentationName, InstrumentationVersion);
+
+            if (options.IsGcEnabled)
+            {
+                this.meter.CreateObservableGauge($"{options.MetricPrefix}gc_heap_size", () => (double)(GC.GetTotalMemory(false) / 1_000_000), "MB", "GC Heap Size");
+                this.meter.CreateObservableGauge($"{options.MetricPrefix}gen_0-gc_count", () => GC.CollectionCount(0), description: "Gen 0 GC Count");
```

Comment:
I would suggest we use a single separator rather than a mixture of both "-" and "_".

Reply:
I would propose int64 `process.runtime.dotnet.gc.heap`, and the unit is `Bytes`.

For GC count, use `process.runtime.dotnet.gc.count`, and put `gen` as dimension/attribute.